Corona-Warn-App/src/test/java/de/rki/coronawarnapp/tracing/ui/details/TracingDetailsItemProviderTest.kt

```diff
@@ -0,0 +1,349 @@
+
+    @MockK lateinit var tracingStatus: GeneralTracingStatus
+    @MockK lateinit var tracingRepository: TracingRepository
+    @MockK lateinit var riskLevelStorage: RiskLevelStorage
+
+    @BeforeEach
+    fun setup() {
+        MockKAnnotations.init(this)
+        every { context.resources } returns resources
+    }
+
+    @AfterEach
+    fun teardown() {
+        clearAllMocks()
+    }
+
+    private fun createInstance() = TracingDetailsItemProvider(
+        tracingStatus = tracingStatus,
+        tracingRepository = tracingRepository,
+        riskLevelStorage = riskLevelStorage
+    )
+
+    private fun prepare(
+        status: GeneralTracingStatus.Status,
+        activeTracingDaysInRetentionPeriod: Long,
+        riskLevel: ProtoRiskLevel,
+        matchedKeyCount: Int
+    ) {
+        every { tracingStatus.generalStatus } returns flowOf(status)
+        every { tracingRepository.activeTracingDaysInRetentionPeriod } returns flowOf(activeTracingDaysInRetentionPeriod)
+        every { aggregatedRiskResult.totalRiskLevel } returns riskLevel
+
+        if (riskLevel == ProtoRiskLevel.LOW) {
+            every { aggregatedRiskResult.isLowRisk() } returns true
+        } else if (riskLevel == ProtoRiskLevel.HIGH) {
+            every { aggregatedRiskResult.isIncreasedRisk() } returns true
+        }
+
+        val exposureWindow: ExposureWindow = mockk()
+
+        val riskLevelResult = RiskLevelTaskResult(
+            calculatedAt = Instant.EPOCH,
+            aggregatedRiskResult = aggregatedRiskResult,
+            exposureWindows = listOf(exposureWindow)
+        )
+        every { riskLevelResult.matchedKeyCount } returns matchedKeyCount
+        every { riskLevelStorage.latestAndLastSuccessful } returns flowOf(listOf(riskLevelResult))
+    }
+
+    @Test
+    fun `additional info low risk box`() = runBlockingTest {
+
+        prepare(
+            status = GeneralTracingStatus.Status.TRACING_ACTIVE,
+            activeTracingDaysInRetentionPeriod = 0,
+            riskLevel = ProtoRiskLevel.LOW,
+            matchedKeyCount = 1
+        )
+
+        val instance = createInstance()
+        val testCollector = instance.state.test(startOnScope = this)
+
+        testCollector.latestValue?.size?.shouldBeGreaterThan(0)
+        testCollector.latestValue?.any { it is AdditionalInfoLowRiskBox.Item } shouldBe true
+    }
+
+    @Test
+    fun `no additional info low risk box due to matched key count`() = runBlockingTest {
+
+        prepare(
+            status = GeneralTracingStatus.Status.TRACING_ACTIVE,
+            activeTracingDaysInRetentionPeriod = 0,
+            riskLevel = ProtoRiskLevel.LOW,
+            matchedKeyCount = 0
+        )
+
+        val instance = createInstance()
+        val testCollector = instance.state.test(startOnScope = this)
+
+        testCollector.latestValue?.size?.shouldBeGreaterThan(0)
+        testCollector.latestValue?.any { it is AdditionalInfoLowRiskBox.Item } shouldBe false
```

Comment:
Please remove any use of `?.` in test assertions, if it should not be null, then it should crash if is null in a test.